testcases/PowerNVDump.py

```diff
@@ -0,0 +1,416 @@
+                done = True
+                boot_type = BootType.INVALID
+                self.cv_SYSTEM.set_state(OpSystemState.UNKNOWN_BAD)
+            except PlatformError:
+                done = True
+                boot_type = BootType.NORMAL
+            if rc == 0:
+                self.cv_SYSTEM.set_state(OpSystemState.IPLing)
+                done = True
+            if rc == 1 or rc == 2:
+                print "Kdump finished collecting core file, waiting for regular IPL to complete"
+
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        #self.cv_HOST.ssh.state = SSHConnectionState.DISCONNECTED
+        print "System booted fine to host OS..."
+        return boot_type
+
+
+class OPALCrash_MPIPL(PowerNVDump):
+
+    def runTest(self):
+        if not self.pdbg or not os.path.exists(self.pdbg):
+            self.fail("pdbg file %s doesn't exist" % self.pdbg)
+        self.setup_test()
+        cmd = "rm /tmp/pdbg; rm /tmp/deadbeef"
+        try:
+            self.cv_BMC.run_command(cmd)
+        except CommandFailed:
+            pass
+        # copy the pdbg file to BMC
+        self.cv_BMC.image_transfer(self.pdbg)
+        deadbeef = os.path.join(self.basedir, "test_binaries", "deadbeef")
+        self.cv_BMC.image_transfer(deadbeef)
+        if "OpenBMC" in self.bmc_type:
+            cmd = "/tmp/pdbg -a putmem 0x300000f8 < /tmp/deadbeef"
+        elif "SMC" in self.bmc_type:
+            cmd = "/tmp/rsync_file/pdbg putmem 0x300000f8 < /tmp/rsync_file/deadbeef"
+        self.cv_BMC.run_command(cmd)
+        done = False
+        boot_type = BootType.NORMAL
+        rc = -1
+        while not done:
+            try:
+                rc = self.c.pty.expect(['\n', 'ISTEP 10.1'], timeout=300)
+            except(SkibootAssert, KernelFADUMP):
+                print "====================MPIPL boot started==================="
+                # if fadump is enabled & kdump is disabled, system should start
+                # MPIPL after kernel crash(oops)
+                self.cv_SYSTEM.set_state(OpSystemState.IPLing)
+                boot_type = BootType.MPIPL
+            except pexpect.TIMEOUT:
+                done = True
+                boot_type = BootType.INVALID
+                self.cv_SYSTEM.set_state(OpSystemState.UNKNOWN_BAD)
+            except PlatformError:
+                done = True
+                boot_type = BootType.NORMAL
+            if rc == 1:
+                self.cv_SYSTEM.set_state(OpSystemState.IPLing)
+                done = True
+
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        #self.cv_HOST.ssh.state = SSHConnectionState.DISCONNECTED
+        print "System booted fine to host OS..."
+        self.verify_dump_file(boot_type)
+        return boot_type
+
+
+class KernelCrash_FadumpEnable(PowerNVDump):
+
+    def runTest(self):
+        self.setup_test()
+        if not self.is_mpipl_supported():
+            self.skipTest("MPIPL support is not found")
```

Comment:
Here we have to validate device tree properties before checking kernel settings.

Reply:
sure. will do.